pkg/resolver/search.go

```diff
@@ -19,6 +20,8 @@ import (
 	"k8s.io/klog/v2"
 )
 
+var mux sync.RWMutex // Mutex to lock map during read/write
```

Comment:
This is global so it's shared by all requests.  I think we should declare it in the `SearchResult` struct so it is scoped to an individual request.